### ISSUE TITLE
Lag when used in NextJs

### ISSUE THREAD

**Issue by timothygachengo** [justin-chu/react-fast-marquee#92]
1. When using the `autoFill` option, the marquee freezes up the page in dev mode when using NextJs 13.

**Comment by thisispaul** [comment]
@timothygachengo just had this problem and looks like I solved it by making the component a client component.

Just add "use client" to the top of the component/page you are using this marquee in.

**Comment by thisispaul** [comment]
Same issue and it actually just seems to completely freeze the page.

**Comment by timothygachengo** [comment]
I tried that and it still doesn't work.

Regards,
Timothy Mugo.
Software Developer.
Phone: +254795114587

**Comment by thisispaul** [comment]
> I tried that and it still doesn't work.

yeah i deleted my comment as it seemed to work but then actually just broke again. sorry for confusion.

**Comment by thisispaul** [comment]
@timothygachengo I fixed it locally by cloning the poackage and changing that whole resize observer this

```
  // Calculate width and multiplier on mount and on window resize
  useEffect(() => {
    if (!isMounted) return;
    calculateWidth();
    window.addEventListener("resize", calculateWidth);
    return () => window.removeEventListener("resize", calculateWidth);
  }, [calculateWidth, containerRef, isMounted]);
```

It's not as ideal as the original code as it will call calculateWidth more but it doesn't have the same issue of infinitely calling itself and breaking the page.

**Comment by wayneslabs** [comment]
I'm not sure if it works for nextjs, but you can try the answer described here: https://github.com/justin-chu/react-fast-marquee/issues/66#issuecomment-1575052529

**Comment by jomarmontuya** [comment]
I've encountered the same issues I totally drop this library and just use this instead as alternative -> https://ui.aceternity.com/components/infinite-moving-cards

**Comment by gregsadetsky** [comment]
seeing this now when using `autoFill` in a React app - when set to true, it create ~38k nodes/copies of the scrolled element

<img width="247" alt="image" src="https://github.com/user-attachments/assets/038e13b9-a752-454f-86ab-d3f5fb3721ce">

thanks to @wayneslabs for pointing out the [answer](https://github.com/justin-chu/react-fast-marquee/issues/66#issuecomment-1575052529): "wrap the Marquee in a div and bound its width to say 500px"

it'd be great to have a check (when autoFill=true) to not create more than... 100? copies of the contents.
